static/app/components/profiling/flamegraphSearch.tsx

```diff
@@ -132,6 +132,7 @@ function FlamegraphSearch({
   canvasPoolManager,
 }: FlamegraphSearchProps): React.ReactElement | null {
   const [search, dispatchSearch] = useFlamegraphSearch();
+  const [didInitialSearch, setDidInitialSearch] = useState(!search.query);
```

Comment:
Would it make more sense to use `useRef` here so that it doesn't trigger extra rerenders?

Reply:
don't think it will make much of a difference, a re-render will happen when we call `handleChange` and `dispatchSearch` is called either way. _also_ we'd only be re-rendering this tiny component.